base/workers_pool.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "base/assert.hpp"
+#include "base/macros.hpp"
+
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <thread>
+
+namespace base
+{
+
+class WorkersPool
+{
+public:
+  class Task
+  {
+  public:
+    virtual ~Task() = default;
+    virtual void Do() = 0;
+  };
+
+  explicit WorkersPool(size_t threadsNum)
```

Comment:
Why another pool? Because the old one was single-threaded?

Reply:
ThreadPool is a bit fancy and designed for work-cancel-stop.